hibernate-core/src/main/java/org/hibernate/query/sqm/sql/BaseSqmToSqlAstConverter.java

```diff
@@ -7105,20 +7100,21 @@ private JdbcMappingContainer getBooleanType() {
 	@Override
 	public Object visitBooleanExpressionPredicate(SqmBooleanExpressionPredicate predicate) {
 		final Expression booleanExpression = (Expression) predicate.getBooleanExpression().accept( this );
-		if ( booleanExpression instanceof SelfRenderingExpression ) {
-			final Predicate sqlPredicate = new SelfRenderingPredicate( (SelfRenderingExpression) booleanExpression );
-			if ( predicate.isNegated() ) {
-				return new NegatedPredicate( sqlPredicate );
-			}
-			return sqlPredicate;
-		}
```

Comment:
I think this code here was to support functions as predicates, though it seems we lack a test for this. Can you please re-add it or do you think the code path would never be hit?

Reply:
I mean, there should be a test...

Reply:
Did not hear a reply, so added 2 basic tests of boolean functions wrt predicates -

1. as a predicate - `... where boolean_function`
2. in a predicate - `... where boolean_function = true`

Both work with the code as-is